### PR TITLE
Added all api tests for accounts folder

### PR DESCRIPTION
**Pull request by YahyoMahmudov**

### CODE REVIEW

File: src/account/getAccount.ts [hunk 1]
```diff
@@ -29,7 +28,131 @@ export async function getAccountFavourites(): Promise<AxiosResponse<AccountDetai
     };
 
     try {
-        const response = await httpRequest<AccountDetailsI>('GET', url, queryParams);
+        const response = await httpRequest<ResponseType.MovieAndTV>('GET', url, queryParams);
+        return response;
```
Comment (jkalandarov) [inline src/account/getAccount.ts]: response object consists of several properties including headers, config, response, status, statusText. Does the user  who calls `getAccountFavouriteMovies` function need to get all those properties? Does the function return favourite movies as the name tells or favourite movies and headers, config, etc?

File: src/account/getAccount.ts [hunk 2]
```diff
@@ -29,7 +28,131 @@ export async function getAccountFavourites(): Promise<AxiosResponse<AccountDetai
...
+        throw error;
+    }
+}
+
+export async function getAccountFavouriteTV(): Promise<AxiosResponse> {
+    const url = `account/${accountId}/favorite/tv`;
+
+    const queryParams = {
+        language: 'en-US',
+        page: 1,
```
Comment (jkalandarov) [inline src/account/getAccount.ts]: what if the user wants to get the results on page 2?

File: src/account/getAccount.ts [hunk 3]
```diff
@@ -29,7 +28,131 @@ export async function getAccountFavourites(): Promise<AxiosResponse<AccountDetai
...
+        sort_by: 'created_at.asc'
+    };
+
+    try {
+        const response = await httpRequest<ResponseType.MovieAndTV>('GET', url, queryParams);
+        return response;
+    } catch (error) {
+        console.error('Error fetching account details:', error);
+        throw error;
```
Comment (jkalandarov) [inline src/account/getAccount.ts]: both console.error and throw error print on the console. Do we need both of them?

File: src/account/postAccount.test.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import * as AccountModule from './postAccount';
+
+describe('Get account favourite movies', () => {
+
+    test('Verify that the status code of the response is 200', async () => {
+        const { status } = await AccountModule.addAccountFavourite();
```
Comment (jkalandarov) [inline src/account/postAccount.test.ts]: describe block tells about getting the account's favorite movies. but you're calling a function that adds account favorite smth? The function name is ambiguous.

File: src/account/postAccount.test.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+        expect(status).toBe(201);
+    });
+});
+
+describe('Get account watchlist movies', () => {
+
+    test('Verify that the status code of the response is 200', async () => {
+        const { status } = await AccountModule.addAccountWatchlist();
```
Comment (jkalandarov) [inline src/account/postAccount.test.ts]: what shows, movies or cartoons are being added to watch list?

File: src/account/getAccount.test.ts [hunk 6]
```diff
@@ -1,23 +1,78 @@
...
-import { getAccountFavourites } from './getAccount';
+import * as AccountModule from './getAccount';
 
 describe('Get account details', () => {
 
     test('Verify that a account id is not null.', async () => {
-        const data = await getAccountDetails();
-        expect(data.data.id).not.toBe(null);
+        const data = await AccountModule.getAccountDetails();
+        expect(data.data.avatar.gravatar).not.toBe(null);
```
Comment (jkalandarov) [inline src/account/getAccount.test.ts]: Test name is about verifying account id.

File: src/account/getAccount.test.ts [hunk 7]
```diff
@@ -1,23 +1,78 @@
...
     });
 });
 
-describe('Get account favourites', () => {
+describe('Get account favourite movies', () => {
 
     test('Verify that the status code of the response is 200', async () => {
-        const { status } = await getAccountFavourites();
+        const { status } = await AccountModule.getAccountFavouriteMovies();
+        expect(status).toBe(200);
```
Comment (jkalandarov) [inline src/account/getAccount.test.ts]: Verify Favourite movies list in the response.

File: src/account/postAccount.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import { AxiosResponse } from 'axios';
+import { httpRequest } from '../httpMethod/httpRequest';
+import * as ResponseType from './accountResponse'
+import dotenv from 'dotenv';
+dotenv.config();
+
+const accountId = process.env.ACCOUNT_ID;
+
+export async function addAccountFavourite(): Promise<AxiosResponse> {
```
Comment (jkalandarov) [inline src/account/postAccount.ts]: The function should accept movie id from the parameters

File: src/account/postAccount.ts [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    try {
+        const response = await httpRequest<ResponseType.Status>('POST', url, body);
+        return response;
+    } catch (error) {
+        console.error('Error fetching account details:', error);
+        throw error;
+    }
+}
+
+export async function addAccountWatchlist(): Promise<AxiosResponse> {
```
Comment (jkalandarov) [inline src/account/postAccount.ts]: The function should accept movie id from the parameters

**Review by jkalandarov — CHANGES_REQUESTED** [review]

File: src/account/getAccount.ts (hunk 2 above)
Comment (YahyoMahmudov) [reply src/account/getAccount.ts]: Fixed

File: src/account/getAccount.ts (hunk 3 above)
Comment (YahyoMahmudov) [reply src/account/getAccount.ts]: Fixed

File: src/account/postAccount.test.ts (hunk 4 above)
Comment (YahyoMahmudov) [reply src/account/postAccount.test.ts]: Fixed

File: src/account/getAccount.test.ts (hunk 7 above)
Comment (YahyoMahmudov) [reply src/account/getAccount.test.ts]: Fixed

File: src/account/postAccount.ts (hunk 8 above)
Comment (YahyoMahmudov) [reply src/account/postAccount.ts]: Fixed

File: src/account/postAccount.ts (hunk 9 above)
Comment (YahyoMahmudov) [reply src/account/postAccount.ts]: Fixed

File: src/account/postAccount.test.ts (hunk 5 above)
Comment (YahyoMahmudov) [reply src/account/postAccount.test.ts]: fixed

File: src/account/getAccount.test.ts (hunk 6 above)
Comment (YahyoMahmudov) [reply src/account/getAccount.test.ts]: fixed

File: src/account/postAccount.test.ts [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import * as postAccount from './postAccount';
+
+describe('Add favourite movie to the account', () => {
+
+    test('Verify that the status code of the response is 201', async () => {
+        const response = await postAccount.addAccountFavourite(390);
```
Comment (jkalandarov) [inline src/account/postAccount.test.ts]: You need to check if the movie or smth with ID 390 was really added.

File: src/account/postAccount.test.ts [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+        expect(response.status).toBe(201);
+    });
+});
+
+describe('Add watchlist movie to the account', () => {
+
+    test('Verify that the status code of the response is 201', async () => {
+        const response = await postAccount.addAccountWatchlist(389);
```
Comment (jkalandarov) [inline src/account/postAccount.test.ts:31]: You need to check watchlist was updated.

**Review by jkalandarov — CHANGES_REQUESTED** [review]

File: src/account/postAccount.test.ts (hunk 10 above)
Comment (YahyoMahmudov) [reply src/account/postAccount.test.ts]: Fixed

File: src/account/postAccount.test.ts (hunk 11 above)
Comment (YahyoMahmudov) [reply src/account/postAccount.test.ts:31]: Fixed

**Review by jkalandarov — APPROVED** [review]

**Pull request merged**